src/dmd/typesem.d

```diff
@@ -1504,3 +1506,548 @@ static Type merge(Type type)
+            }
+        }
+        else if (ident == Id.min_exp)
+        {
+            switch (mt.ty)
+            {
+            case Tcomplex32:
+            case Timaginary32:
+            case Tfloat32:
+                ivalue = Target.FloatProperties.min_exp;
+                goto Lint;
+            case Tcomplex64:
+            case Timaginary64:
+            case Tfloat64:
+                ivalue = Target.DoubleProperties.min_exp;
+                goto Lint;
+            case Tcomplex80:
+            case Timaginary80:
+            case Tfloat80:
+                ivalue = Target.RealProperties.min_exp;
+                goto Lint;
+            default:
+                break;
+            }
+        }
+        visit(cast(Type)mt);
+        return;
+
+    Livalue:
+        e = new IntegerExp(loc, ivalue, mt);
+        result = e;
+        return;
+
+    Lfvalue:
+        if (mt.isreal() || mt.isimaginary())
+            e = new RealExp(loc, fvalue, mt);
+        else
+        {
+            const cvalue = complex_t(fvalue, fvalue);
+            //for (int i = 0; i < 20; i++)
+            //    printf("%02x ", ((unsigned char *)&cvalue)[i]);
+            //printf("\n");
+            e = new ComplexExp(loc, cvalue, mt);
+        }
+        result = e;
+        return;
+
+    Lint:
+        e = new IntegerExp(loc, ivalue, Type.tint32);
+        result = e;
+    }
+
+    override void visit(TypeVector mt)
+    {
+        visit(cast(Type)mt);
+    }
+
+    override void visit(TypeEnum mt)
+    {
+        Expression e;
+        if (ident == Id.max || ident == Id.min)
+        {
+            result = mt.sym.getMaxMinValue(loc, ident);
+            return;
+        }
+        else if (ident == Id._init)
+        {
+            e = mt.defaultInitLiteral(loc);
+        }
+        else if (ident == Id.stringof)
+        {
+            const s = mt.toChars();
+            e = new StringExp(loc, cast(char*)s);
+            Scope sc;
+            e = e.expressionSemantic(&sc);
+        }
+        else if (ident == Id._mangleof)
+        {
+            visit(cast(Type)mt);
+            e = result;
+        }
+        else
+        {
+            e = mt.toBasetype().getProperty(loc, ident, flag);
+        }
+        result = e;
+    }
+
+    override void visit(TypeTuple mt)
+    {
+        Expression e;
+        static if (LOGDOTEXP)
+        {
+            printf("TypeTuple::getProperty(type = '%s', ident = '%s')\n", mt.toChars(), ident.toChars());
+        }
+        if (ident == Id.length)
+        {
+            e = new IntegerExp(loc, mt.arguments.dim, Type.tsize_t);
+        }
+        else if (ident == Id._init)
+        {
+            e = mt.defaultInitLiteral(loc);
+        }
+        else if (flag)
+        {
+            e = null;
+        }
+        else
+        {
+            error(loc, "no property `%s` for tuple `%s`", ident.toChars(), mt.toChars());
+            e = new ErrorExp();
+        }
+        result = e;
+    }
+}
+
+
```

Comment:
Trailing lines at the end.

Reply:
I think that the right place for this is in typesem since getProperty, dotExp and resolve are all part of the semantic phase.